dsp/src/trig.rs

```diff
@@ -41,39 +48,40 @@ pub fn atan2(y: i32, x: i32) -> i32 {
     //
     // which is taken from Rajan 2006: Efficient Approximations for
     // the Arctangent Function.
-    let (min, max) = if ux < uy { (ux, uy) } else { (uy, ux) };
+    let (min, max) = if y_greater { (x, y) } else { (y, x) };
```

Comment:
Now that you don't use `y` and `x` anymore below, just

```suggestion
    if y_greater {
        core::mem::swap(&mut x, &mut y);
     }
```

and keep using those.